File: flutter_module/lib/features/wifi/domain/entities/wifi_network.dart
class WifiNetwork {
  final String ssid;
  final String bssid;
  final String capabilities;
  final int level; // RSSI-like
  final int frequency;
  final bool isSecured;
  final bool isSaved;
  final bool isConnected;

  const WifiNetwork({
    required this.ssid,
    required this.bssid,
    this.capabilities = '',
    required this.level,
    required this.frequency,
    this.isSecured = true,
    this.isSaved = false,
    this.isConnected = false,
  });
}

class SavedWifiNetwork {
  final String ssid;
  final String password;
  final int priority;
  final bool autoConnect;
  final DateTime savedTimestamp;

  const SavedWifiNetwork({
    required this.ssid,
    required this.password,
    this.priority = 0,
    this.autoConnect = true,
    required this.savedTimestamp,
  });
}

enum WifiState { disconnected, connecting, connected, disconnecting, suspended, error, disabled }

class WifiConnectionStatus {
  final WifiState state;
  final WifiNetwork? network;
  final String? ipAddress;
  final String? message;

  const WifiConnectionStatus({
    required this.state,
    this.network,
    this.ipAddress,
    this.message,
  });
}

File: app/src/main/java/com/example/androidapp/data/model/WifiNetwork.kt
package com.example.androidapp.data.model

data class WifiNetwork(
    val ssid: String,
    val bssid: String,
    val capabilities: String,
    val level: Int,
    val frequency: Int,
    val isSecured: Boolean = true,
    val isSaved: Boolean = false,
    val isConnected: Boolean = false
)

data class SavedWifiNetwork(
    val ssid: String,
    val password: String,
    val priority: Int = 0,
    val autoConnect: Boolean = true,
    val savedTimestamp: Long = System.currentTimeMillis()
)

data class WifiConnectionStatus(
    val state: WifiState,
    val network: WifiNetwork?,
    val ipAddress: String? = null,
    val message: String? = null
)

enum class WifiState {
    DISABLED,
    DISCONNECTED,
    CONNECTING,
    CONNECTED,
    SUSPENDED,
    DISCONNECTING,
    ERROR
}